pkg/cmd/infra/router/f5.go

```diff
@@ -230,9 +228,18 @@ func (o *F5RouterOptions) Run() error {
 		return err
 	}
 
-	statusPlugin := controller.NewStatusAdmitter(f5Plugin, routeclient.Route(), o.RouterName, "")
-	uniqueHostPlugin := controller.NewUniqueHost(statusPlugin, o.RouteSelectionFunc(), o.RouterSelection.DisableNamespaceOwnershipCheck, statusPlugin)
-	plugin := controller.NewHostAdmitter(uniqueHostPlugin, o.F5RouteAdmitterFunc(), false, o.RouterSelection.DisableNamespaceOwnershipCheck, statusPlugin)
+	var recorder controller.RejectionRecorder = controller.LogRejections
+	var plugin router.Plugin = f5Plugin
+	if o.UpdateStatus {
+		status := controller.NewStatusAdmitter(plugin, routeclient.Route(), o.RouterName, o.RouterCanonicalHostname)
+		recorder = status
+		plugin = status
+	}
+	if o.ExtendedValidation {
+		plugin = controller.NewExtendedValidator(plugin, recorder)
+	}
+	plugin = controller.NewUniqueHost(plugin, o.RouteSelectionFunc(), o.RouterSelection.DisableNamespaceOwnershipCheck, recorder)
+	plugin = controller.NewHostAdmitter(plugin, o.F5RouteAdmitterFunc(), o.AllowWildcardRoutes, o.RouterSelection.DisableNamespaceOwnershipCheck, recorder)
```

Comment:
`false` instead of `o.AllowWildcardRoutes`
I didn't think we supported wildcard routes on the F5 router? AFAICR, it was doing an exact host match in the lookup. That said my knowledge is a bit dated - so maybe that's changed -  @Miciah  / @rajatchopra can confirm.

Reply:
No, we explicitly block wildcard routes on F5.  https://github.com/openshift/origin/blob/master@%7B2018-02-20%7D/pkg/cmd/infra/router/f5.go#L194

Reply:
Yeah, I didn't want the code to drift.  If we add it, it can be done in the admitter func.

Reply:
Can understand that. Am ok with the change but just wanted to call out here  that if someone accidentally sets an environment variable to allow wildcard routes to the F5 router, it may cause different behavior (well untested previously) in the host admitter `https://github.com/smarterclayton/origin/blob/3ac3cd0562bf2f3cc55f80500d6ed410813d1402/pkg/router/controller/host_admitter.go#L131`.

Reply:
We are planning to deprecate the F5 Router pretty soon, anyway. So its a no factor.

Reply:
Deprecate or remove?

Reply:
Remove eventually. It will stay until the field is ready to go fully with the container connector solution by F5.
The deprecation notice will be up in 3.9 docs.